### PR TITLE
Isosurface opacity

### PR DESCRIPTION
**Pull request by archmoj**
This PR enables `opacity` attribute for `isosurface` traces.
Also it addressed bug #3546 and removed the second drawings to find the edges [see here](https://github.com/plotly/plotly.js/blob/b94dccee250e26dcab828a1c8e0e726b72db9cd3/test/image/baselines/gl3d_isosurface_transparent.png).

In addition some improvements such as reduction of tolerance (to remove glitch in mesh generation #3547) are made in this PR. Various functions now return `true` or `false` result in case triangles are generated. This information is stored in the first pass & could be used in second pass processes.

This PR would be helpful for the `volume` PR  #3488.
@plotly/plotly_js 
 
[Before codepen](https://codepen.io/MojtabaSamimi/pen/mvzOXd?editors=0010).
[After codepen](https://codepen.io/MojtabaSamimi/pen/bzmBgE?editors=0010).

### CODE REVIEW

**Comment by etpinard** [comment]
@archmoj Is this one ready to review?

File: src/traces/isosurface/attributes.js [hunk 1]
```diff
@@ -235,10 +235,11 @@ colorscaleAttrs('', {
 
     colorbar: colorbarAttrs,
 
-    contour: meshAttrs.contour,
+    opacity: meshAttrs.opacity,
```
Comment (etpinard) [inline src/traces/isosurface/attributes.js]: We should probably point out in the mesh3d attributes that the current behavior isn't perfect and may be subject to change in the (near) future cc https://github.com/plotly/plotly.js/issues/3243

**Comment by archmoj** [comment]
> @archmoj Is this one ready to review?

Yes please. @etpinard I thank you in advance!

**Comment by etpinard** [comment]
Looks good! 

https://github.com/plotly/plotly.js/pull/3545#discussion_r257390424 is my only comment.

**Comment by archmoj** [comment]
> Looks good!
> 
> [#3545 (comment)](https://github.com/plotly/plotly.js/pull/3545#discussion_r257390424) is my only comment.

@etpinard Thanks for the review.
In respect to your comment, it is done in https://github.com/plotly/plotly.js/pull/3545/commits/abc8ae25fe0f9e92c51f2a85847383ba8ea06aa8.

**Comment by etpinard** [comment]
Thanks @archmoj !

:dancer:

**Pull request merged**